src/main/java/Listener.java

```diff
@@ -21,12 +21,30 @@ public void update() throws Exception {
         Conlin.getHomework(Conlin.getMonth() + Conlin.getYear());
     }
 
+    private String getClosestHWDate() {
```

Comment:
@VincentZhao12 Ideally this should go back to the previous month, because January 1st edge cases probably suck.

Also note that you should be careful here and as someone suggested, put the previous day (i.e. until today morning put 9/15 homework).